packages/jupyterlab-voila/src/preview.ts

```diff
@@ -0,0 +1,43 @@
+import { MainAreaWidget, IFrame, ToolbarButton } from "@jupyterlab/apputils";
+
+export const VOILA_ICON_CLASS = "jp-MaterialIcon jp-VoilaIcon";
+
+export namespace VoilaPreview {
+  export interface IOptions extends MainAreaWidget.IOptionsOptionalContent {
+    url: string;
+    label: string;
+  }
+}
+
+export class VoilaPreview extends MainAreaWidget<IFrame> {
+  constructor(options: VoilaPreview.IOptions) {
+    let { url, label, ...opts } = options;
+    super({
+      ...opts,
+      content: new IFrame()
+    });
+
+    this.content.url = url;
+    this.content.title.label = label;
+    this.content.title.icon = VOILA_ICON_CLASS;
+    this.content.id = `voila-${++Private.count}`;
```

Comment:
What is it for this `id`?

Reply:
It's used as the DOM id for the phosphor widget